// boxrec-requests/dist/boxrec-requests.constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeightDivision = exports.BoxrecRole = exports.BoxrecFighterOption = exports.BoxrecFighterRole = exports.WeightDivisionCapitalized = exports.Country = exports.Sport = exports.BoxingBoutOutcome = exports.WinLossDraw = exports.BoxrecStatus = void 0;
var BoxrecStatus;
(function (BoxrecStatus) {
    BoxrecStatus["active"] = "a";
    BoxrecStatus["all"] = "";
})(BoxrecStatus = exports.BoxrecStatus || (exports.BoxrecStatus = {}));
var WinLossDraw;
(function (WinLossDraw) {
    WinLossDraw["win"] = "win";
    WinLossDraw["loss"] = "loss";
    WinLossDraw["draw"] = "draw";
    WinLossDraw["scheduled"] = "scheduled";
    WinLossDraw["unknown"] = "unknown";
})(WinLossDraw = exports.WinLossDraw || (exports.WinLossDraw = {}));
var BoxingBoutOutcome;
(function (BoxingBoutOutcome) {
    BoxingBoutOutcome["TKO"] = "technical knockout";
    BoxingBoutOutcome["KO"] = "knockout";
    BoxingBoutOutcome["UD"] = "unanimous decision";
    BoxingBoutOutcome["MD"] = "majority decision";
    BoxingBoutOutcome["SD"] = "split decision";
    BoxingBoutOutcome["TD"] = "technical decision";
    BoxingBoutOutcome["RTD"] = "corner retirement";
    BoxingBoutOutcome["DQ"] = "disqualification";
    BoxingBoutOutcome["NWS"] = "newspaper decision";
})(BoxingBoutOutcome = exports.BoxingBoutOutcome || (exports.BoxingBoutOutcome = {}));
var Sport;
(function (Sport) {
    Sport["proBoxing"] = "Pro Boxing";
    Sport["worldSeries"] = "World Series";
})(Sport = exports.Sport || (exports.Sport = {}));
// if needed to rebuild
// grab the dropdown HTML from the locations page
// run it through regexr with this regex
// `value\s?\=\s?\"(\w{1,2})\"\s?(?:selected\=\"selected\")?\>\s?([A-zÀ-ÿ\s’\-\(\)\.]+)`
// go to `list` and use this `"$2" = "$1",\n`
var Country;
(function (Country) {
    Country["Afghanistan"] = "AF";
    Country["Albania"] = "AL";
    Country["Algeria"] = "DZ";
    Country["American Samoa"] = "AS";
    Country["Andorra"] = "AD";
    Country["Angola"] = "AO";
    Country["Anguilla"] = "AI";
    Country["Antarctica"] = "AQ";
    Country["Antigua And Barbuda"] = "AG";
    Country["Argentina"] = "AR";
    Country["Armenia"] = "AM";
    Country["Aruba"] = "AW";
    Country["Australia"] = "AU";
    Country["Austria"] = "AT";
    Country["Azerbaijan"] = "AZ";
    Country["Bahamas"] = "BS";
    Country["Bahrain"] = "BH";
    Country["Bangladesh"] = "BD";
    Country["Barbados"] = "BB";
    Country["Belarus"] = "BY";
    Country["Belgium"] = "BE";
    Country["Belize"] = "BZ";
    Country["Benin"] = "BJ";
    Country["Bermuda"] = "BM";
    Country["Bhutan"] = "BT";
    Country["Bolivia"] = "BO";
    Country["Bonaire"] = "BQ";
    Country["Bosnia And Herzegovina"] = "BA";
    Country["Botswana"] = "BW";
    Country["Bouvet Island"] = "BV";
    Country["Brazil"] = "BR";
    Country["British Indian Ocean Territory"] = "IO";
    Country["British Virgin Islands"] = "VG";
    Country["Brunei"] = "BN";
    Country["Bulgaria"] = "BG";
    Country["Burkina Faso"] = "BF";
    Country["Burundi"] = "BI";
    Country["Cambodia"] = "KH";
    Country["Cameroon"] = "CM";
    Country["Canada"] = "CA";
    Country["Cape Verde"] = "CV";
    Country["Cayman Islands"] = "KY";
    Country["Central African Republic"] = "CF";
    Country["Chad"] = "TD";
    Country["Chile"] = "CL";
    Country["China"] = "CN";
    Country["Christmas Island"] = "CX";
    Country["Cocos (Keeling) Islands"] = "CC";
    Country["Colombia"] = "CO";
    Country["Comoros"] = "KM";
    Country["Congo"] = "CG";
    Country["Cook Islands"] = "CK";
    Country["Costa Rica"] = "CR";
    Country["Croatia"] = "HR";
    Country["Cuba"] = "CU";
    Country["Cura\u00E7ao"] = "CW";
    Country["Cyprus"] = "CY";
    Country["Czech Republic"] = "CZ";
    Country["C\u00F4te D\u2019Ivoire"] = "CI";
    Country["Democratic Republic Of The Congo"] = "CD";
    Country["Denmark"] = "DK";
    Country["Djibouti"] = "DJ";
    Country["Dominica"] = "DM";
    Country["Dominican Republic"] = "DO";
    Country["Ecuador"] = "EC";
    Country["Egypt"] = "EG";
    Country["El Salvador"] = "SV";
    Country["Equatorial Guinea"] = "GQ";
    Country["Eritrea"] = "ER";
    Country["Estonia"] = "EE";
    Country["Ethiopia"] = "ET";
    Country["Falkland Islands"] = "FK";
    Country["Faroe Islands"] = "FO";
    Country["Fiji"] = "FJ";
    Country["Finland"] = "FI";
    Country["France"] = "FR";
    Country["French Guiana"] = "GF";
    Country["French Polynesia"] = "PF";
    Country["French Southern Territories"] = "TF";
    Country["Gabon"] = "GA";
    Country["Gambia"] = "GM";
    Country["Georgia"] = "GE";
    Country["Germany"] = "DE";
    Country["Ghana"] = "GH";
    Country["Gibraltar"] = "GI";
    Country["Greece"] = "GR";
    Country["Greenland"] = "GL";
    Country["Grenada"] = "GD";
    Country["Guadeloupe"] = "GP";
    Country["Guam"] = "GU";
    Country["Guatemala"] = "GT";
    Country["Guinea"] = "GN";
    Country["Guinea-Bissau"] = "GW";
    Country["Guyana"] = "GY";
    Country["Haiti"] = "HT";
    Country["Heard Island And Mcdonald Islands"] = "HM";
    Country["Honduras"] = "HN";
    Country["Hong Kong S.A.R."] = "HK";
    Country["Hungary"] = "HU";
    Country["Iceland"] = "IS";
    Country["India"] = "IN";
    Country["Indonesia"] = "ID";
    Country["Iran"] = "IR";
    Country["Iraq"] = "IQ";
    Country["Ireland"] = "IE";
    Country["Israel"] = "IL";
    Country["Italy"] = "IT";
    Country["Jamaica"] = "JM";
    Country["Japan"] = "JP";
    Country["Jordan"] = "JO";
    Country["Kazakhstan"] = "KZ";
    Country["Kenya"] = "KE";
    Country["Kiribati"] = "KI";
    Country["Kosovo"] = "XK";
    Country["Kuwait"] = "KW";
    Country["Kyrgyzstan"] = "KG";
    Country["Laos"] = "LA";
    Country["Latvia"] = "LV";
    Country["Lebanon"] = "LB";
    Country["Lesotho"] = "LS";
    Country["Liberia"] = "LR";
    Country["Libya"] = "LY";
    Country["Liechtenstein"] = "LI";
    Country["Lithuania"] = "LT";
    Country["Luxembourg"] = "LU";
    Country["Macao S.A.R."] = "MO";
    Country["Macedonia"] = "MK";
    Country["Madagascar"] = "MG";
    Country["Malawi"] = "MW";
    Country["Malaysia"] = "MY";
    Country["Maldives"] = "MV";
    Country["Mali"] = "ML";
    Country["Malta"] = "MT";
    Country["Marshall Islands"] = "MH";
    Country["Martinique"] = "MQ";
    Country["Mauritania"] = "MR";
    Country["Mauritius"] = "MU";
    Country["Mayotte"] = "YT";
    Country["Mexico"] = "MX";
    Country["Micronesia"] = "FM";
    Country["Moldova"] = "MD";
    Country["Monaco"] = "MC";
    Country["Mongolia"] = "MN";
    Country["Montenegro"] = "ME";
    Country["Montserrat"] = "MS";
    Country["Morocco"] = "MA";
    Country["Mozambique"] = "MZ";
    Country["Myanmar"] = "MM";
    Country["Namibia"] = "NA";
    Country["Nauru"] = "NR";
    Country["Nepal"] = "NP";
    Country["Netherlands"] = "NL";
    Country["Netherlands Antilles"] = "AN";
    Country["New Caledonia"] = "NC";
    Country["New Zealand"] = "NZ";
    Country["Nicaragua"] = "NI";
    Country["Niger"] = "NE";
    Country["Nigeria"] = "NG";
    Country["Niue"] = "NU";
    Country["Norfolk Island"] = "NF";
    Country["North Korea"] = "KP";
    Country["Northern Mariana Islands"] = "MP";
    Country["Norway"] = "NO";
    Country["Oman"] = "OM";
    Country["Pakistan"] = "PK";
    Country["Palau"] = "PW";
    Country["Palestinian Territory"] = "PS";
    Country["Panama"] = "PA";
    Country["Papua New Guinea"] = "PG";
    Country["Paraguay"] = "PY";
    Country["Peru"] = "PE";
    Country["Philippines"] = "PH";
    Country["Pitcairn"] = "PN";
    Country["Poland"] = "PL";
    Country["Portugal"] = "PT";
    Country["Puerto Rico"] = "PR";
    Country["Qatar"] = "QA";
    Country["Romania"] = "RO";
    Country["Russia"] = "RU";
    Country["Rwanda"] = "RW";
    Country["R\u00E9union"] = "RE";
    Country["Saint Helena"] = "SH";
    Country["Saint Kitts And Nevis"] = "KN";
    Country["Saint Lucia"] = "LC";
    Country["Saint Pierre And Miquelon"] = "PM";
    Country["Saint Vincent And The Grenadines"] = "VC";
    Country["Samoa"] = "WS";
    Country["San Marino"] = "SM";
    Country["Sao Tome And Principe"] = "ST";
    Country["Saudi Arabia"] = "SA";
    Country["Senegal"] = "SN";
    Country["Serbia"] = "RS";
    Country["Seychelles"] = "SC";
    Country["Sierra Leone"] = "SL";
    Country["Singapore"] = "SG";
    Country["Sint Maarten"] = "SX";
    Country["Slovakia"] = "SK";
    Country["Slovenia"] = "SI";
    Country["Solomon Islands"] = "SB";
    Country["Somalia"] = "SO";
    Country["South Africa"] = "ZA";
    Country["South Georgia And The South Sandwich Islands"] = "GS";
    Country["South Korea"] = "KR";
    Country["South Sudan"] = "SS";
    Country["Spain"] = "ES";
    Country["Sri Lanka"] = "LK";
    Country["Sudan"] = "SD";
    Country["Suriname"] = "SR";
    Country["Svalbard And Jan Mayen"] = "SJ";
    Country["Swaziland"] = "SZ";
    Country["Sweden"] = "SE";
    Country["Switzerland"] = "CH";
    Country["Syria"] = "SY";
    Country["Taiwan"] = "TW";
    Country["Tajikistan"] = "TJ";
    Country["Tanzania"] = "TZ";
    Country["Thailand"] = "TH";
    Country["Timor-Leste"] = "TL";
    Country["Togo"] = "TG";
    Country["Tokelau"] = "TK";
    Country["Tonga"] = "TO";
    Country["Trinidad And Tobago"] = "TT";
    Country["Tunisia"] = "TN";
    Country["Turkey"] = "TR";
    Country["Turkmenistan"] = "TM";
    Country["Turks And Caicos Islands"] = "TC";
    Country["Tuvalu"] = "TV";
    Country["U.S. Virgin Islands"] = "VI";
    Country["Uganda"] = "UG";
    Country["Ukraine"] = "UA";
    Country["United Arab Emirates"] = "AE";
    Country["United Kingdom"] = "UK";
    Country["United States Minor Outlying Islands"] = "UM";
    Country["Uruguay"] = "UY";
    Country["USA"] = "US";
    Country["Uzbekistan"] = "UZ";
    Country["Vanuatu"] = "VU";
    Country["Vatican"] = "VA";
    Country["Venezuela"] = "VE";
    Country["Vietnam"] = "VN";
    Country["Wallis And Futuna"] = "WF";
    Country["Western Sahara"] = "EH";
    Country["Yemen"] = "YE";
    Country["Yugoslavia"] = "YU";
    Country["Zambia"] = "ZM";
    Country["Zimbabwe"] = "ZW";
})(Country = exports.Country || (exports.Country = {}));
// the params for searching titles are capitalized divisions
var WeightDivisionCapitalized;
(function (WeightDivisionCapitalized) {
    WeightDivisionCapitalized["heavyweight"] = "Heavyweight";
    WeightDivisionCapitalized["cruiserweight"] = "Cruiserweight";
    WeightDivisionCapitalized["lightHeavyweight"] = "Light Heavyweight";
    WeightDivisionCapitalized["superMiddleweight"] = "Super Middleweight";
    WeightDivisionCapitalized["middleweight"] = "Middleweight";
    WeightDivisionCapitalized["superWelterweight"] = "Super Welterweight";
    WeightDivisionCapitalized["welterweight"] = "Welterweight";
    WeightDivisionCapitalized["superLightweight"] = "Super Lightweight";
    WeightDivisionCapitalized["lightweight"] = "Lightweight";
    WeightDivisionCapitalized["superFeatherweight"] = "Super Featherweight";
    WeightDivisionCapitalized["featherweight"] = "Featherweight";
    WeightDivisionCapitalized["superBantamweight"] = "Super Bantamweight";
    WeightDivisionCapitalized["bantamweight"] = "Bantamweight";
    WeightDivisionCapitalized["superFlyweight"] = "Super Flyweight";
    WeightDivisionCapitalized["flyweight"] = "Flyweight";
    WeightDivisionCapitalized["lightFlyweight"] = "Light Flyweight";
    WeightDivisionCapitalized["minimumweight"] = "Minimumweight";
})(WeightDivisionCapitalized = exports.WeightDivisionCapitalized || (exports.WeightDivisionCapitalized = {}));
var BoxrecFighterRole;
(function (BoxrecFighterRole) {
    BoxrecFighterRole["proBoxer"] = "proboxer";
    BoxrecFighterRole["amateurBoxer"] = "amateurboxer";
    BoxrecFighterRole["amateurKickBoxer"] = "amateurkickboxer";
    BoxrecFighterRole["proKickBoxer"] = "kickboxer";
    BoxrecFighterRole["muayThaiBoxer"] = "muaythaiboxer";
    BoxrecFighterRole["amateurMuayThaiBoxer"] = "amateurmuaythaiboxer";
    BoxrecFighterRole["bareKnuckleBoxer"] = "bareknuckleboxer";
    BoxrecFighterRole["worldSeriesBoxer"] = "worldseriesboxer";
})(BoxrecFighterRole = exports.BoxrecFighterRole || (exports.BoxrecFighterRole = {}));
// used as params for `locations/events`
var BoxrecFighterOption;
(function (BoxrecFighterOption) {
    BoxrecFighterOption["Pro Boxing"] = "proboxing";
    BoxrecFighterOption["Amateur Boxing"] = "amateurboxing";
    BoxrecFighterOption["Pro Kickboxing"] = "kickboxing";
    BoxrecFighterOption["Amateur Kickboxing"] = "amateurkickboxing";
    BoxrecFighterOption["Pro Muay Thai Boxing"] = "amateurmuaythaiboxing";
    BoxrecFighterOption["Bare Knuckle Boxing"] = "bareknuckleboxing";
    BoxrecFighterOption["World Series Boxing"] = "worldseriesboxing";
})(BoxrecFighterOption = exports.BoxrecFighterOption || (exports.BoxrecFighterOption = {}));
var BoxrecRole;
(function (BoxrecRole) {
    BoxrecRole["proBoxer"] = "proboxer";
    BoxrecRole["amateurBoxer"] = "amateurboxer";
    BoxrecRole["amateurKickBoxer"] = "amateurkickboxer";
    BoxrecRole["kickboxer"] = "kickboxer";
    BoxrecRole["proKickBoxer"] = "kickboxer";
    BoxrecRole["muayThaiBoxer"] = "muaythaiboxer";
    BoxrecRole["proMuayThaiBoxer"] = "muaythaiboxer";
    BoxrecRole["amateurMuayThaiBoxer"] = "amateurmuaythaiboxer";
    BoxrecRole["bareKnuckleBoxer"] = "bareknuckleboxer";
    BoxrecRole["doctor"] = "doctor";
    BoxrecRole["inspector"] = "inspector";
    BoxrecRole["judge"] = "judge";
    BoxrecRole["manager"] = "manager";
    BoxrecRole["matchmaker"] = "matchmaker";
    BoxrecRole["promoter"] = "promoter";
    BoxrecRole["referee"] = "referee";
    BoxrecRole["supervisor"] = "supervisor";
})(BoxrecRole = exports.BoxrecRole || (exports.BoxrecRole = {}));
var WeightDivision;
(function (WeightDivision) {
    WeightDivision["heavyweight"] = "heavyweight";
    WeightDivision["cruiserweight"] = "cruiserweight";
    WeightDivision["lightHeavyweight"] = "light heavyweight";
    WeightDivision["superMiddleweight"] = "super middleweight";
    WeightDivision["middleweight"] = "middleweight";
    WeightDivision["superWelterweight"] = "super welterweight";
    WeightDivision["welterweight"] = "welterweight";
    WeightDivision["superLightweight"] = "super lightweight";
    WeightDivision["lightweight"] = "lightweight";
    WeightDivision["superFeatherweight"] = "super featherweight";
    WeightDivision["featherweight"] = "featherweight";
    WeightDivision["superBantamweight"] = "super bantamweight";
    WeightDivision["bantamweight"] = "bantamweight";
    WeightDivision["superFlyweight"] = "super flyweight";
    WeightDivision["flyweight"] = "flyweight";
    WeightDivision["lightFlyweight"] = "light flyweight";
    WeightDivision["minimumweight"] = "minimumweight";
})(WeightDivision = exports.WeightDivision || (exports.WeightDivision = {}));
